src/Raven.Server/RavenServer.cs

```diff
@@ -929,45 +935,188 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
+                                                       $"Registering the new certificate explicitly. Security Clearance: {newCertDef.SecurityClearance}, " +
+                                                       $"Permissions:{Environment.NewLine}{string.Join(Environment.NewLine, newCertDef.Permissions.Select(kvp => kvp.Key + ": " + kvp.Value.ToString()))}");
+
+                                cert = ctx.ReadObject(newCertDef.ToJson(), "Client/Certificate/Definition");
+                            }
+                            else
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.UnfamiliarCertificate;
+                            }
                         }
-                        else
+
+                        if (cert != null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Allowed;
-                            foreach (var kvp in definition.Permissions)
+                            var definition = JsonDeserializationServer.CertificateDefinition(cert);
+                            authenticationStatus.Definition = definition;
+                            if (definition.SecurityClearance == SecurityClearance.ClusterAdmin)
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
+                            }
+                            else if (definition.SecurityClearance == SecurityClearance.ClusterNode)
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
+                            }
+                            else if (definition.SecurityClearance == SecurityClearance.Operator)
                             {
-                                authenticationStatus.AuthorizedDatabases.Add(kvp.Key, kvp.Value);
+                                authenticationStatus.Status = AuthenticationStatus.Operator;
+                            }
+                            else
+                            {
+                                authenticationStatus.Status = AuthenticationStatus.Allowed;
+                                foreach (var kvp in definition.Permissions)
+                                {
+                                    authenticationStatus.AuthorizedDatabases.Add(kvp.Key, kvp.Value);
+                                }
                             }
                         }
                     }
+                    finally
+                    {
+                        ctx.CloseTransaction();
+                    }
                 }
             }
 
             return authenticationStatus;
         }
 
+        private bool CertHasKnownIssuer(X509Certificate2 certificate, List<CertificateDefinition> certsWithSameHash)
```

Comment:
Chain of server and user must be:

* Same length
* Same order
* Public key hash same for all

Reply:
```
for(var i = 0; i < userChain.Length; i++)
{
    if(userChain[i].PublicKeyHash != serverChain[i].PublicKeyHash)
         return false;
}
```

Reply:
Alternatively, the user chain issuer is in one of the well known issuers.

Reply:
done. (Comparing against a known good client cert and not the server cert)